### ISSUE TITLE
Ember client hangs on large responses when F[_] is a Kleisli

### ISSUE THREAD

**Issue by gatorcse** [http4s/http4s#3556]
I was experimenting with integrating the http4s client with Natchez tracing, and I ran into a bug with Ember. When I converted the `F[_]` to be the `Kleisli[....]` that Natchez needs, the ember client started to hang on requests that were more than one chunk. I don't know if it's failing to terminate after the stream is over, or if it is failing to proceed past the second chunk; I've only been able to test it with 2 chunk responses.

### Example Code
_EmberBroken.scala_
```scala
package emberbroken

import cats._
import cats.data.Kleisli
import cats.implicits._
import cats.effect._
import cats.effect.implicits._
import natchez.{EntryPoint, Span, Trace}
import org.http4s._
import org.http4s.client.blaze.BlazeClientBuilder
import org.http4s.implicits._
//import org.http4s.client.blaze.BlazeClientBuilder
import org.http4s.ember.client.EmberClientBuilder

import scala.concurrent.ExecutionContext

object EmberBroken extends IOApp {
  // This will always work
//  override def run(args: List[String]): IO[ExitCode] = runIO
  
  // This will fail when the stream has multiple chunks
  override def run(args: List[String]): IO[ExitCode] = runTraced

  def runIO = runF[IO]
  def runTraced = entryPoint[IO].use { ep =>
    ep.root("Testing Ember App").use { root =>
      runF[Kleisli[IO, Span[IO], *]].run(root)
    }
  }

  def runF[F[_]: Concurrent: Timer: ContextShift]: F[ExitCode] =
    EmberClientBuilder.default[F].build
    .use { c =>
      c.get((uri"https://baconipsum.com/api/")
        .withQueryParam("type", "meat-and-filler")
        .withQueryParam("paras", "100") // Set this to something smaller and it works
        .withQueryParam("format", "text")
      ) {
        res => Sync[F].delay { println("Got a response") } *>
          res.bodyText
            .evalTap(text => Sync[F].delay { println("Have a chunk")})
            .compile.drain
      }
        .flatTap(_ => Sync[F].delay { println("fetched") }) // Not printed when response is too large
        .as(ExitCode.Success)
    }

  def entryPoint[F[_]: Sync]: Resource[F, EntryPoint[F]] = {
    import natchez.jaeger.Jaeger
    import io.jaegertracing.Configuration.SamplerConfiguration
    import io.jaegertracing.Configuration.ReporterConfiguration
    Jaeger.entryPoint[F]("natchez-example") { c =>
      Sync[F].delay {
        c.withSampler(SamplerConfiguration.fromEnv)
          .withReporter(ReporterConfiguration.fromEnv)
          .getTracer
      }
    }
  }
}
```

_build.sbt_
```scala
name := "emberbroken"

scalaVersion := "2.13.2" // Also supports 2.12.x

val http4sVersion = "0.21.6"

// Only necessary for SNAPSHOT releases
resolvers += Resolver.sonatypeRepo("snapshots")

//scalacOptions += "-Ypartial-unification"

libraryDependencies += "org.http4s" %% "http4s-dsl" % http4sVersion
libraryDependencies += "org.http4s" %% "http4s-blaze-client" % http4sVersion
libraryDependencies += "org.http4s" %% "http4s-ember-client" % http4sVersion
libraryDependencies += "org.tpolecat"  %% "natchez-jaeger" % "0.0.11"

addCompilerPlugin("org.typelevel" %% "kind-projector" % "0.11.0" cross CrossVersion.full)
```

**Comment by gatorcse** [comment]
Actually, I accidentally let it hang for a minute or so and it eventually completed.

**Comment by kevinmeredith** [comment]
From where is `ep` coming, @gatorcse? I'm assuming it's `entryPoint`?

**Comment by gatorcse** [comment]
yeah, entryPoint gives a `Resource` of `EntryPoint`. The `entryPoint` code is copied directly from the natchez examples

**Comment by rossabaker** [comment]
Needs a retest, because Ember has seen a lot of progress since then.

**Comment by gatorcse** [comment]
k, I'll check when I can, thanks!

**Comment by gatorcse** [comment]
@rossabaker Fixed, seems to be fixed as of `0.21.7`, the very next version after my initial test. Also confirmed works in 0.22.0-M7
